design-system/src/theme-provider.tsx

```diff
@@ -28,31 +27,35 @@ type ThemeProviderProps = {
 };
 
 const ThemeProvider = (props: ThemeProviderProps) => {
-  const root = useRef<HTMLDivElement>();
+  const root = document.querySelector(':root') as HTMLElement;
   const [theme, setTheme] = useState<ThemeName>('default');
 
   const changeTheme = useCallback((newTheme: string) => {
-    if (allThemesNames.some((themeName) => themeName === newTheme)) {
-      setTheme(newTheme as ThemeName);
-    } else {
-      setTheme('default');
-    }
+    const isNewThemeValid = allThemesNames.some(
+      (themeName) => themeName === newTheme
+    );
```

Comment:
nit: since we are iterating over a string array, I guess it would be simpler (even faster) to use `.include()` instead:
```
const isNewThemeValid = allThemesNames.include(newTheme);
setTheme(isNewThemeValid ? (newTheme as ThemeName) : 'default');
warning(
  isNewThemeValid
  `ThemeProvider: the specified theme '${newTheme}' is not supported.`
);
```